src/Tasks/AssemblyDependency/ReferenceTable.cs

```diff
@@ -407,6 +407,12 @@ internal void AddReference(AssemblyNameExtension assemblyName, Reference referen
                 }
             }
 
+            if (reference.FullPath.Length > 0 && ChangeWaves.AreFeaturesEnabled(ChangeWaves.Wave17_0))
+            {
+                // Saves effort and makes deduplication possible downstream
+                reference.FullPath = FileUtilities.NormalizePath(reference.FullPath);
```

Comment:
Setting the `FullPath` prop has a side-effect of re-running the `IsWinMDFile` check. Do you think it would be worth optimizing? Perhaps by introducing a "NormalizeFullPath" method on `Reference`.

Reply:
Sounds reasonable to me! I only did it in one case because the other case had previously set FullPath explicitly, so it was intentionally running the check.

Reply:
Thank you! I'm curious about these statements:

https://github.com/dotnet/msbuild/blob/d150e93ff1f8828bb8b0b5fda64f4582d61a7e73/src/Tasks/AssemblyDependency/Reference.cs#L491-L493

They were executed in the previous version. Now you set only `_fullPath` and leave these fields unchanged. Intentional?

Reply:
I don't think _fileNameWithoutExtension should care, and I added the other two. Looking through the rest of the method, if the _fullPath was not null or empty before, this shouldn't make it null or empty, so I think that should be unchanged. Whether something is a winmd file or not also shouldn't care if the path is normalized or not.